arviz/data/inference_data.py

```diff
@@ -324,6 +378,322 @@ def sel(
+
+        Parameters
+        ----------
+        dim: Hashable or iterable of Hashable, optional
+            Dimension(s) over which to unstack. By default unstacks all MultiIndexes.
+        groups: str or list of str, optional
+            Groups where the selection is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        inplace: bool, optional
+            If ``True``, modify the InferenceData object inplace,
+            otherwise, return the modified copy.
+
+        Returns
+        -------
+        InferenceData
+            A new InferenceData object by default.
+            When `inplace==True` perform selection in place and return `None`
+
+        """
+
+        groups = self._group_names(groups, filter_groups)
+        if isinstance(dim, str):
+            dim = [dim]
+
+        out = self if inplace else deepcopy(self)
+        for group in groups:
+            dataset = getattr(self, group)
+            valid_dims = set(dim).intersection(dataset.dims) if dim is not None else dim
+            dataset = dataset.unstack(dim=valid_dims)
+            setattr(out, group, dataset)
+        if inplace:
+            return None
+        else:
+            return out
+
+    def rename(self, name_dict=None, groups=None, filter_groups=None, inplace=False):
+        """Perform xarray renaming of variable and dimensions on all groups.
+
+        Loops groups to perform Dataset.rename(name_dict)
+        for every key in name_dict if key is a dimension/data_vars of the dataset.
+        The renaming is performed on all relevant groups (like
+        posterior, prior, sample stats) while non relevant groups like observed data are
+        omitted. See :meth:`xarray:xarray.Dataset.rename`
+
+        Parameters
+        ----------
+        name_dict: dict
+            Dictionary whose keys are current variable or dimension names
+            and whose values are the desired names.
+        groups: str or list of str, optional
+            Groups where the selection is to be applied. Can either be group names
+            or metagroup names.
+        filter_groups: {None, "like", "regex"}, optional, default=None
+            If `None` (default), interpret groups as the real group or metagroup names.
+            If "like", interpret groups as substrings of the real group or metagroup names.
+            If "regex", interpret groups as regular expressions on the real group or
+            metagroup names. A la `pandas.filter`.
+        inplace: bool, optional
+            If ``True``, modify the InferenceData object inplace,
```

Comment:
```suggestion
            If ``True``, modify the InferenceData object in-place,
```

Should we use in-place in all occurrences?